pkg/3scale/amp/operator/zync_reconciler.go

```diff
@@ -44,69 +50,102 @@ func (r *ZyncReconciler) Reconcile() (reconcile.Result, error) {
+		zyncQueMutators = append(zyncQueMutators, reconcilers.DeploymentReplicasMutator)
 	}
-	err = r.ReconcileDeploymentConfig(zync.QueDeploymentConfig(), reconcilers.DeploymentConfigMutator(zyncQueMutators...))
+	err = r.ReconcileDeployment(zync.QueDeployment(ampImages.Options.ZyncImage), reconcilers.DeploymentMutator(zyncQueMutators...))
 	if err != nil {
 		return reconcile.Result{}, err
 	}
 
+	// 3scale 2.14 -> 2.15
+	isMigrated, err = upgrade.MigrateDeploymentConfigToDeployment(component.ZyncQueDeploymentName, r.apiManager.GetNamespace(), false, r.Client(), r.BaseReconciler.Scheme())
+	if err != nil {
+		return reconcile.Result{}, err
+	}
+	if !isMigrated {
+		return reconcile.Result{Requeue: true}, nil
+	}
+
+	serviceMutators := []reconcilers.MutateFn{
+		reconcilers.CreateOnlyMutator,
+		reconcilers.ServiceSelectorMutator,
+	}
+
 	// Zync Service
-	err = r.ReconcileService(zync.Service(), reconcilers.CreateOnlyMutator)
+	err = r.ReconcileService(zync.Service(), reconcilers.ServiceMutator(serviceMutators...))
 	if err != nil {
 		return reconcile.Result{}, err
 	}
 
 	if !r.apiManager.IsExternal(appsv1alpha1.ZyncDatabase) {
-		// Zync DB DC
-		zyncDBDCMutator := reconcilers.DeploymentConfigMutator(
-			reconcilers.DeploymentConfigImageChangeTriggerMutator,
-			reconcilers.DeploymentConfigContainerResourcesMutator,
-			reconcilers.DeploymentConfigAffinityMutator,
-			reconcilers.DeploymentConfigTolerationsMutator,
-			reconcilers.DeploymentConfigPodTemplateLabelsMutator,
-			reconcilers.DeploymentConfigPriorityClassMutator,
-			reconcilers.DeploymentConfigTopologySpreadConstraintsMutator,
-			reconcilers.DeploymentConfigPodTemplateAnnotationsMutator,
+		// Zync DB Deployment
+		zyncDBDMutator := reconcilers.DeploymentMutator(
+			reconcilers.DeploymentContainerResourcesMutator,
+			reconcilers.DeploymentAffinityMutator,
+			reconcilers.DeploymentTolerationsMutator,
+			reconcilers.DeploymentPodTemplateLabelsMutator,
+			reconcilers.DeploymentPriorityClassMutator,
+			reconcilers.DeploymentTopologySpreadConstraintsMutator,
+			reconcilers.DeploymentPodTemplateAnnotationsMutator,
+			reconcilers.DeploymentPodContainerImageMutator,
 		)
-		err = r.ReconcileDeploymentConfig(zync.DatabaseDeploymentConfig(), zyncDBDCMutator)
+		err = r.ReconcileDeployment(zync.DatabaseDeployment(ampImages.Options.ZyncDatabasePostgreSQLImage), zyncDBDMutator)
 		if err != nil {
 			return reconcile.Result{}, err
 		}
 
+		// 3scale 2.14 -> 2.15
+		isMigrated, err = upgrade.MigrateDeploymentConfigToDeployment(component.ZyncDatabaseDeploymentName, r.apiManager.GetNamespace(), false, r.Client(), nil)
+		if err != nil {
+			return reconcile.Result{}, err
+		}
+		if !isMigrated {
+			return reconcile.Result{Requeue: true}, nil
+		}
+
 		// Zync DB Service
-		err = r.ReconcileService(zync.DatabaseService(), reconcilers.CreateOnlyMutator)
+		err = r.ReconcileService(zync.DatabaseService(), reconcilers.ServiceMutator(serviceMutators...))
```

Comment:
Q
Similar to one of the above questions, create only vs expected/desired reconciliation. Why can/must we do this?

Reply:
Same as above, the Service selector used to reference [DeploymentConfigs](https://github.com/3scale/3scale-operator/blame/de98237e57d7a68a4f44c5a27edabeebd8184136/pkg/3scale/amp/component/zync.go#L537) but it now needs to reference [Deployments](https://github.com/3scale/3scale-operator/blob/845cddc4d9cc1405c6f10b65f50c727063e438d5/pkg/3scale/amp/component/zync.go#L493).